src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs

```diff
@@ -100,7 +108,15 @@ private async ValueTask SelectItemAsync(CancellationToken cancellationToken)
         {
             // Switch to the UI so we can determine where the user is and determine the state the last time we updated
             // the UI.
-            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+#pragma warning disable VSTHRD004 // Await SwitchToMainThreadAsync
```

Comment:
Should we push to have the analyzer changed to recognize this pattern?

Here, right? [https://github.com/microsoft/vs-threading/blob/main/src/Microsoft.VisualStudio.Threading.Analyzers/AbstractVSTHRD004AwaitSwitchToMainThreadAsyncAnalyzer.cs](https://github.com/microsoft/vs-threading/blob/main/src/Microsoft.VisualStudio.Threading.Analyzers/AbstractVSTHRD004AwaitSwitchToMainThreadAsyncAnalyzer.cs)

Reply:
Yes, I filed a report previously and the issue has already been fixed in code.

Reply:
Looks great, thanks!